### DISCUSSION TITLE
Observed classes

### DISCUSSION THREAD

**Discussion by gintriago12** [online-ml/river#716] [Q&A] [answered]
Hi,

Where does a model such as Logistic Regression, or the ensemble ADWINBaggingClassifier store the observed classes?

The Hoeffding Adaptive Tree stores the observed clases in the attribute classes. However, the models mentioned first do not have such attribute.

Thanks for your help.

Best,

**Comment by raphaelsty** [comment, marked as answer]
Hi @gintriago12, `Logistic Regression` or `ADWINBaggingClassifier` do not store the observed classes. `Logistic Regression` and `ADWINBaggingClassifier` do not needs such attribute to learn or predict. If you need to know the frequency of the target variable during training, you can use [collections.Counter](https://docs.python.org/fr/3/library/collections.html#collections.Counter) in the training loop. 🙂

**Reply by gintriago12** [reply]
Thanks Raphaelsty!

**Reply by MaxHalford** [reply]
Also, `LogisticRegression` is a binary classifier, so the classes are implicitely `False` and `True`.
